### PR TITLE
[stable/anchore-engine] Split environment variables configMap

### LINKED ISSUE #20174: [stable/anchore-engine] envFrom anchore-engine configMap
**Describe the bug**
The deployments set the environment from the `anchore-engine` `configMap`. The `configMap` contains the environment variables as well as the `config.yaml` resulting in the `config.yaml` being set in the environment:
Screenshot from 2020-01-15 16-10-46

**Version of Helm and Kubernetes**:
`helm` version:
```
Client: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
```
`kubectl` version:
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.4-gke.22", GitCommit:"a6ba43f5a24ac29e631bb627c9b2a719c4e93638", GitTreeState:"clean", BuildDate:"2019-11-26T00:40:25Z", GoVersion:"go1.12.11b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:
```
name: anchore-engine
version: 1.4.0
```

**What happened**:
Environment created from `config.yaml` (visible in `env`).

**What you expected to happen**:
`config.yaml` not in the containers environment.

**How to reproduce it** (as minimally and precisely as possible):
`helm install --name anchore-engine --namespace anchore-engine stable/anchore-engine`

**Anything else we need to know**:
Def. not a blocking issue. I [split the environment variable templates into a new `configMap`](https://github.com/masseybradley/charts/commit/3b0f8fd32c029927592e72a0dddb728a722dcdae)
 and using the `envFrom` with that `configMap` which I'd like to PR.

### PR DESCRIPTION
**Pull request by masseybradley**
Signed-off-by: Bradley Massey <bradley.massey@yields.io>

#### What this PR does / why we need it:

Splits the environment variables and `config.yaml` configMaps for non-enterprise deployments.

#### Which issue this PR fixes

  - fixes #20174

#### Special notes for your reviewer:

This PR does not address the issue in the enterprise deployment.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by Btodhunter** [comment]
@masseybradley thanks for the PR! I saw in the issue that you mentioned removing the enterprise-feeds changes. Could you add those again? It would be nice to take care of all this env configmap stuff at once. 

I will pull and test your PR before approving to make sure it doesn't break anything w/ enterprise installs.

**Comment by masseybradley** [comment]
Hi @Btodhunter - thanks again. I added a commit to address the `enterprise-feeds` configMap environment variables split (I think it should be good).

File: stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{{- if and .Values.anchoreEnterpriseGlobal.enabled (or .Values.anchoreEnterpriseRbac.enabled .Values.anchoreEnterpriseReports.enabled) -}}
```
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml]: this should be `{{- if and .Values.anchoreEnterpriseGlobal.enabled .Values.anchoreEnterpriseFeeds.enabled }}`

File: stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+{{- $component := "enterprise" -}}
```
Comment (Btodhunter) [inline stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml]: the component for this is "enterprise-feeds"

**Review by Btodhunter — CHANGES_REQUESTED** [review]

File: stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml (hunk 2 above)
Comment (masseybradley) [reply stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml]: Thanks for catching those :+1:

File: stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml (hunk 1 above)
Comment (masseybradley) [reply stable/anchore-engine/templates/enterprise_feeds_configmap_env.yaml]: Thanks. I kept the end of line whitespace chomp `... -}}`.

**Comment by masseybradley** [comment]
@Btodhunter should I rewrite the history for my commits?

**Comment by Btodhunter** [comment]
whoops - I lgtm right as you asked that question. I think it's okay, your commits will get squashed before merging.

**Comment by masseybradley** [comment]
> @masseybradley whoops - I lgtm right as you asked that question. I think it's okay, your commits will get squashed before merging.

Is cool. Was just wondering if there were any strict guidelines or preferences, didn't see anything in the [`CONTRIBUTING.md`](https://github.com/helm/charts/blob/master/CONTRIBUTING.md).

**Pull request merged**